code/datums/components/payment.dm

```diff
@@ -0,0 +1,62 @@
+/**
+  * Handles simple payment operations where the cost of the object in question doesn't change.
+  *
+  * What this is useful for:
+  * Basic forms of vending.
+  * Objects that can drain the owner's money linearly.
+  * What this is not useful for:
+  * Things where the seller may want to fluxuate the price of the object.
+  * Improving standardizing every form of payment handing, as some custom handling is specific to that object.
+  **/
+/datum/component/payment
+	///Standardized of operation.
+	var/cost = 10
```

Comment:
you can have dynamic pricing by just allowing to send a signal to override the charged amount, and hence make the component a lot more flexible

Reply:
in other words another signal that takes a 'cost' parameter which is used to attempt to charge the account with that cost

Reply:
Oh yeah, that could very easily be done going forward, I was just hoping to put it off until more cases of the simple vendor can be worked out.